homeassistant/components/nextbus/config_flow.py

```diff
@@ -0,0 +1,264 @@
+"""Config flow to configure the Nextbus integration."""
+import logging
+
+from py_nextbus import NextBusClient
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import (
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import CONF_AGENCY, CONF_ROUTE, CONF_STOP, DOMAIN
+from .util import UserConfig
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _dict_to_select_selector(options: dict[str, str]) -> SelectSelector:
+    return SelectSelector(
+        SelectSelectorConfig(
+            options=[
+                SelectOptionDict(value=key, label=value)
+                for key, value in options.items()
+            ],
+            mode=SelectSelectorMode.DROPDOWN,
+        )
+    )
+
+
+class NextBusFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle Nextbus configuration."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize NextBus config flow."""
+        self.nextbus_config: dict[str, str] = {}
+        self._client = NextBusClient(output_format="json")
+        self._agency_tags: dict[str, str] = {}
+        self._route_tags: dict[str, str] = {}
+        self._stop_tags: dict[str, str] = {}
+        _LOGGER.info("Init new config flow")
+
+    async def _update_agency_config_options(self) -> None:
+        def job():
+            self._agency_tags = {
+                a["tag"]: a["title"] for a in self._client.get_agency_list()["agency"]
+            }
+
+        await self.hass.async_add_executor_job(job)
```

Comment:
I would make this a simple non async helper function outside the config flow.
Then I would call it when needed inside the config flow, wrapped with `async_add_executor_job`.

```python
def _get_agency_tags(client: NextBusClient) -> dict[str, str]:
    """Get agency tags."""
    return {
        a["tag"]: a["title"] for a in self._client.get_agency_list()["agency"]
    }
```